### PR TITLE
Convert the CodeField to Functional Component with Typescript Support

### PR DESCRIPTION
**Pull request by meabed**
- Convert the CustomTestInput  to Functional Component with Typescript Support
- Add the build to github - apart from npm - some people use github to install :) like me.
- Added forward ref on the input 
- Adjust a bit prettier config.
Thank you for the awesome work :) if this PR does't go =) i am happy to contribute a specific change only.
PS: I use the component on production and its great! thank you.

### CODE REVIEW

File: src/TextInputCustom.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const TextInputCustom: React.ForwardRefExoticComponent<React.PropsWithoutRef<ITextInputCustomProps> & React.RefAttributes<unknown>> = forwardRef(
+  (props: ITextInputCustomProps, ref): ReactElement => {
+    const { onPress, editable } = props;
+    // This hack needs to get click position and then calculate what cell on clicked
+    const _onPress = (event: GestureResponderEvent) => {
+      if (onPress && (editable || editable === undefined)) {
+        onPress(event);
+      }
+    };
+    return <TextInput ref={ref} {...props} editable={editable} onTouchStart-={_onPress} />;
```
Comment (retyui) [inline src/TextInputCustom.tsx]: Is it valid js?
```
onTouchStart-={_onPress}
```

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,3 @@
 node_modules/
 coverage/
-lib/
+.idea/
```
Comment (retyui) [inline .gitignore]: Please use a global gitignore for your environment

**~/.gitconfig**
```
[core]
	excludesfile = ~/.gitignore_global
```

**~/.gitignore_global**
```
.idea/
```

Comment (meabed) [reply .gitignore:3]: oops :) its fixed :)

**Comment by meabed** [comment]
its ready if you are :)

**Comment by retyui** [comment]
I will check it for 0.62

File: src/TextInputCustom.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+const TextInputCustom: React.ForwardRefExoticComponent<React.PropsWithoutRef<ITextInputCustomProps> & React.RefAttributes<unknown>> = forwardRef(
+  (props: ITextInputCustomProps, ref): ReactElement => {
+    const { onPress, editable } = props;
+    // This hack needs to get click position and then calculate what cell on clicked
+    const _onPress = (event: GestureResponderEvent) => {
+      if (onPress && (editable || editable === undefined)) {
+        onPress(event);
+      }
+    };
+    return <TextInput ref={ref} {...props} editable={editable} onTouchStart={_onPress} />;
```
Comment (retyui) [inline src/TextInputCustom.tsx:20]: onTouchStart never invoked on Android  RN@0.62

Comment (meabed) [reply src/TextInputCustom.tsx:20]: Yes it’s there for compatibility

Comment (retyui) [reply src/TextInputCustom.tsx:20]: I want to say that is a problem that 

> onTouchStart never invoked on Android RN@0.62

**Pull request closed without merging**